lib/libc/machine/arm/Make.defs

```diff
@@ -60,6 +60,6 @@ else ifeq ($(CONFIG_ARCH_CORTEXM4),y)   # Cortex-M4 is ARMv7E-M
 include ${TOPDIR}/../lib/libc/machine/arm/armv7-m/Make.defs
 else ifeq ($(CONFIG_ARCH_CORTEXM7),y)   # Cortex-M7 is ARMv7E-M
 include ${TOPDIR}/../lib/libc/machine/arm/armv7-m/Make.defs
-else ifeq ($(CONFIG_ARCH_CORTEXM33),y)   # Cortex-M33 is ARMv7E-M
+else ifeq ($(CONFIG_ARCH_ARMV8M_FAMILY),y)   # Cortex-M33/M55 is ARMv8-M
```

Comment:
Let's add this in arch change

Reply:
Lets do this for all 3 lib/libc/* files in the apps commit
Moving all 3 to the arch support commit.